pkg/controller/controller_utils.go

```diff
@@ -216,6 +216,34 @@ func NewControllerExpectations() *ControllerExpectations {
 	return &ControllerExpectations{cache.NewTTLStore(ExpKeyFunc, ExpectationsTimeout)}
 }
 
+// RCControlInterface is an interface that knows how to add or delete
+// replication controllers, as well as increment or decrement them. It is used
+// by the deployment controller to ease testing of actions that it takes.
+// @TODO Decide if we want to use this or use the kube client directly.
+type RCControlInterface interface {
```

Comment:
+1 to using the `client.Interface` directly; the client fake framework is sufficiently advanced now to support all the function mocking capabilities this sort of narrow interface facilitates, IMO.